### PR TITLE
[batch] use a multipart reader/writer to create batches

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

**Comment by jigold** [comment]
@akotlar The context for this is we need to buffer requests made to the server so Konrad can submit 500+ jobs. Otherwise, we were hitting the max msg size.

FYI: @konradjk

**Comment by jigold** [comment]
https://docs.aiohttp.org/en/stable/multipart.html

File: batch/batch/aioclient.py [hunk 1]
```diff
@@ -318,16 +322,29 @@ def create_job(self, image, command=None, args=None, env=None, ports=None,
...
+            if self.callback:
+                batch_doc['callback'] = self.callback
+            mpwriter.append_json(batch_doc)
+
+            docs = []
+            n = 0
+            for jdoc in self._job_docs:
+                n += 1
+                docs.append(jdoc)
+                if n == job_array_size:
```
Comment (akotlar) [inline batch/batch/aioclient.py]: This doesn't seem like it guarantees that we stay under request limit, but I guess now that we don't have a limit specified for client_max_body_size, this is fine?

File: batch/batch/server/server.py [hunk 2]
```diff
@@ -840,30 +840,42 @@ def is_successful(self):
 
 @routes.post('/api/v1alpha/batches/create')
 @authenticated_users_only
-async def create_batch(request, userdata):
-    parameters = await request.json()
+async def create_batch2(request, userdata):
+    reader = await request.multipart()
+
+    part = await reader.next()
+    batch_parameters = await part.json()
```
Comment (akotlar) [inline batch/batch/server/server.py]: I think this is clever

File: batch/batch/aioclient.py [hunk 3]
```diff
@@ -367,6 +384,11 @@ def __init__(self, session, url=None, token_file=None, token=None, headers=None)
             self.url + path, params=params, cookies=self._cookies, headers=self._headers)
         return await response.json()
 
+    async def _post_data(self, path, data=None):
+        response = await self._session.post(
+            self.url + path, data=data, cookies=self._cookies, headers=self._headers)
```
Comment (akotlar) [inline batch/batch/aioclient.py]: You're passing many more parameters here and using a slightly different method ([than here](https://github.com/hail-is/hail/pull/6403/files#diff-48d48b7c081100ca3f369054a604fe36L330)), wondering why

File: batch/batch/aioclient.py (hunk 1 above)
Comment (jigold) [reply batch/batch/aioclient.py]: Konrad was still seeing errors this morning even after that change. I figured we'd successfully sent batches of jobs with 81 jobs through ci, so 50 seemed like a safe number.

File: batch/batch/aioclient.py (hunk 3 above)
Comment (jigold) [reply batch/batch/aioclient.py]: It should be the same. I copied `_post` and didn't change the function signature except data instead of json.

Comment (akotlar) [reply batch/batch/aioclient.py]: Yep, I'm being stupid

**Pull request merged**